### PR TITLE
Enables adding a custom timeout for requests. Fixes bug for current server timeout role attribute

### PR DESCRIPTION
**Pull request by luispresuelVenafi**
- Enables custom timeout in requests using current `server_timeout` attribute for role entry.
- Fixes bug that would lead to unexpected value be saved into role entry while using  server_timeout attribute.
- Exposes `server_timeout` attribute in response when role attributes are displayed in read operation.

### CODE REVIEW

File: plugin/pki/path_venafi_cert_enroll.go [hunk 1]
```diff
@@ -934,6 +934,10 @@ func formRequest(reqData requestData, role *roleEntry, cl *endpoint.Connector, s
 		}
 	}
 
+	if role.ServerTimeout != 0 {
```
Comment (marcos-albornoz) [inline plugin/pki/path_venafi_cert_enroll.go:937]: According to a similar code in the VCert required change, this should be changed to `role.ServerTimeout > 0`

File: plugin/pki/test_env.go [hunk 2]
```diff
@@ -341,6 +342,10 @@ func (e *testEnv) writeRoleToBackendWithData(t *testing.T, configString venafiCo
 		roleData["service_generated_cert"] = data.serviceGeneratedCert
 	}
 
+	if data.serverTimeout != 0 {
```
Comment (marcos-albornoz) [inline plugin/pki/test_env.go:345]: similar situation as above

File: plugin/pki/vcert.go [hunk 3]
```diff
@@ -111,5 +116,27 @@ func (b *backend) getConfig(ctx context.Context, req *logical.Request, role *rol
 		return nil, fmt.Errorf("failed to get Venafi issuer client: %s", err)
 	}
 
+	if role.ServerTimeout != 0 {
```
Comment (marcos-albornoz) [inline plugin/pki/vcert.go:119]: same situation

**Review by rvelaVenafi — APPROVED** [review]

File: go.mod [hunk 4]
```diff
@@ -3,10 +3,10 @@ module github.com/Venafi/vault-pki-backend-venafi
 go 1.21
 
 require (
-	github.com/Venafi/vcert/v5 v5.1.1
+	github.com/Venafi/vcert/v5 v5.4.1-0.20240228184021-dd6dc944a40f
```
Comment (rvelaVenafi) [inline go.mod:6]: Make sure this is pointing to the correct released version and not a commit

**Review by rvelaVenafi — CHANGES_REQUESTED** [review]

File: go.mod (hunk 4 above)
Comment (luispresuelVenafi) [reply go.mod:6]: Totally

File: plugin/pki/path_venafi_cert_enroll.go (hunk 1 above)
Comment (luispresuelVenafi) [reply plugin/pki/path_venafi_cert_enroll.go]: solved by latest commit

**Review by marcos-albornoz — APPROVED** [review]

File: plugin/pki/vcert.go [hunk 5]
```diff
@@ -2,7 +2,12 @@ package pki
 
 import (
 	"context"
+	"crypto/tls"
```
Comment (rvelaVenafi) [inline plugin/pki/vcert.go:5]: imports are not go-imported

**Review by rvelaVenafi — CHANGES_REQUESTED** [review]

File: plugin/pki/vcert.go (hunk 5 above)
Comment (luispresuelVenafi) [reply plugin/pki/vcert.go:5]: fixed by latest commit

**Review by rvelaVenafi — APPROVED** [review]

**Pull request merged**